harvest/trader/tester.py

```diff
@@ -101,7 +101,11 @@ def setup(self, source, interval, aggregations=None, path=None):
         self.aggregations = aggregations if not aggregations == None else []
 
         self.broker.setup(self.watch, interval, self, self.main)
-        self.streamer.setup(self.watch, interval, self, self.main)
+        if self.broker != self.streamer:
+            # Only call the streamer setup if it is a different
+            # instance than the broker otherwise some brokers can
+            # fail!
+            self.streamer.setup(self.watch, interval, self, self.main)
```

Comment:
Ensures that if the streamer and broker are the same instance the setup function is only called once. This is a problem in Alpaca because the websocket listener is started in the setup and calling setup twice will create two websocket listeners which creates errors.